packages/alchemy/src/react/hooks/useClientActions.ts

```diff
@@ -0,0 +1,135 @@
+import { useMutation } from "@tanstack/react-query";
+import { useCallback } from "react";
+import type { Chain, Client, Transport } from "viem";
+import type { SupportedAccounts } from "../../config";
+import type { UseSmartAccountClientResult } from "./useSmartAccountClient";
+
+export type UseClientActionsProps<
+  TTransport extends Transport = Transport,
+  TChain extends Chain | undefined = Chain | undefined,
+  TActions extends { [x: string]: (...args: any[]) => unknown } = {
+    [x: string]: (...args: any[]) => unknown;
+  }
+> = {
+  client?: UseSmartAccountClientResult<
+    TTransport,
+    TChain,
+    SupportedAccounts
+  >["client"];
+  actions: (client: Client<TTransport, TChain, SupportedAccounts>) => TActions;
+};
+
+export type UseClientActionsResult<
+  TActions extends { [x: string]: (...args: any[]) => unknown } = {
+    [x: string]: (...args: any[]) => unknown;
+  }
+> = {
+  executeAction: <TFunctionName extends ExecutableFunctionName<TActions>>(
+    params: ClientActionParameters<TActions, TFunctionName>
+  ) => void;
+  isExecutingAction: boolean;
+};
+
+export type ExecutableFunctionName<
+  TActions extends { [x: string]: (...args: any[]) => unknown } = {
+    [x: string]: (...args: any[]) => unknown;
+  }
+> = keyof TActions extends infer functionName extends string
+  ? [functionName] extends [never]
+    ? string
+    : functionName
+  : string;
+
+export type ExecutableFunctionArgs<
+  TActions extends { [x: string]: (...args: any[]) => unknown } = {
+    [x: string]: (...args: any[]) => unknown;
+  },
+  TFunctionName extends ExecutableFunctionName<TActions> = ExecutableFunctionName<TActions>
+> = Parameters<TActions[TFunctionName]>;
+
+// All of this is based one how viem's `encodeFunctionData` works
+export type ClientActionParameters<
+  TActions extends { [x: string]: (...args: any[]) => unknown } = {
+    [x: string]: (...args: any[]) => unknown;
+  },
+  TFunctionName extends ExecutableFunctionName<TActions> = ExecutableFunctionName<TActions>,
+  allArgs = ExecutableFunctionArgs<
+    TActions,
+    TFunctionName extends ExecutableFunctionName<TActions>
+      ? TFunctionName
+      : ExecutableFunctionName<TActions>
+  >
+> = {
+  functionName: TFunctionName;
+  args: allArgs;
+};
```

Comment:
this is only meant to be used for modular accounts? then we should name this `useModularAccountClient`.
Also, we should also include checking if the plugin is installed first and in general, throw more informative state error back to the caller.

+ if this is only meant to be used for plugin actions, should name the hook accordingly.
Makes sense to group them and put in the subdir of hooks for modular account client actions.

Reply:
No this is meant to be used with any client decorator, not just plugins. You can use the lightAccountActions decorator with this for example